benchmarks/time/src/main/scala/strawman/collection/immutable/HashSetBenchmark.scala

```diff
@@ -21,48 +21,121 @@ class HashSetBenchmark {
 
   var xs: HashSet[Long] = _
   var xss: scala.Array[HashSet[Long]] = _
+  var zipped: HashSet[(Long, Long)] = _
   var randomIndices: scala.Array[Int] = _
+  var randomIndices2: scala.Array[Int] = _
+  var randomXss: scala.Array[HashSet[Long]] = _
 
   @Setup(Level.Trial)
   def initData(): Unit = {
     def freshCollection() = HashSet((1 to size).map(_.toLong): _*)
     xs = freshCollection()
-//    xss = scala.Array.fill(1000)(freshCollection())
-//    if (size > 0) {
-//      randomIndices = scala.Array.fill(1000)(scala.util.Random.nextInt(size))
-//    }
+    xss = scala.Array.fill(1000)(freshCollection())
+    zipped = xs.map(x => (x, x))
+    if (size > 0) {
+      randomIndices = scala.Array.fill(1000)(scala.util.Random.nextInt(size))
+      randomIndices2 = scala.Array.fill(1000)(scala.util.Random.nextInt(size))
+      randomXss = scala.Array.fill(1000)(freshCollection().take(scala.util.Random.nextInt(size)))
+    }
   }
 
   @Benchmark
-//  @OperationsPerInvocation(size)
-  def cons(bh: Blackhole): Unit = {
+  def incl(bh: Blackhole): Unit = {
     var ys = HashSet.empty[Long]
     var i = 0L
     while (i < size) {
       ys = ys + i
-      i = i + 1
+      i += 1
     }
     bh.consume(ys)
   }
 
   @Benchmark
-  def uncons(bh: Blackhole): Unit = bh.consume(xs.tail)
+  def concat(bh: Blackhole): Unit = bh.consume(xs ++ xs)
 
   @Benchmark
-  def concat(bh: Blackhole): Unit = bh.consume(xs ++ xs)
+  def tail(bh: Blackhole): Unit = bh.consume(xs.tail)
 
   @Benchmark
-  def foreach(bh: Blackhole): Unit = {
-    xs.foreach(x => bh.consume(x))
+  def init(bh: Blackhole): Unit = bh.consume(xs.init)
+
+  @Benchmark
+  def loop_foreach(bh: Blackhole): Unit = xs.foreach(x => bh.consume(x))
+
+  @Benchmark
+  def loop_headTail(bh: Blackhole): Unit = {
+    var ys = xs
+    while (ys.nonEmpty) {
+      bh.consume(ys.head)
+      ys = ys.tail
+    }
+  }
+
+  @Benchmark
+  def loop_initLast(bh: Blackhole): Unit = {
+    var ys = xs
+    while (ys.nonEmpty) {
+      bh.consume(ys.last)
+      ys = ys.init
+    }
+  }
+
+  @Benchmark
+  def loop_iterator(bh: Blackhole): Any = {
+    var n = 0
+    val it = xs.iterator()
+    while (it.hasNext) {
+      bh.consume(it.next())
+      n += 1
+    }
+    bh.consume(n)
+  }
+
+  @Benchmark
+  @OperationsPerInvocation(1000)
+  def contains(bh: Blackhole): Unit = {
+    var i = 0
+    while (i < 1000) {
+      bh.consume(xss(i)(i))
```

Comment:
It would be clearer to actually write this as `contains`.  Why require maintainers to remember that sets are also functions `A => Boolean`?